cloud-consumption/src/main/java/com/sequenceiq/consumption/endpoint/ConsumptionInternalV1Controller.java

```diff
@@ -63,23 +62,68 @@ public void unscheduleStorageConsumptionCollection(@AccountId String accountId,
             @NotNull @ValidCrn(resource = {CrnResourceDescriptor.ENVIRONMENT, CrnResourceDescriptor.DATALAKE}) String monitoredResourceCrn,
             @NotEmpty String storageLocation, @ValidCrn(resource = { CrnResourceDescriptor.USER, CrnResourceDescriptor.MACHINE_USER })
             @InitiatorUserCrn @NotEmpty String initiatorUserCrn) {
-        LOGGER.info("Unregistering storage consumption collection for resource with CRN [{}] and location [{}]", monitoredResourceCrn, storageLocation);
-        Optional<Consumption> consumptionOpt =
-                consumptionService.findStorageConsumptionByMonitoredResourceCrnAndLocation(monitoredResourceCrn, storageLocation);
-        consumptionOpt.ifPresent(consumption -> {
-            jobService.unschedule(consumption);
-            consumptionService.delete(consumption);
-        });
+        LOGGER.info("Unregistering storage consumption collection for resource with CRN [{}] and location [{}]",
+                monitoredResourceCrn, storageLocation);
+        unscheduleConsumption(monitoredResourceCrn, storageLocation);
     }
 
     @Override
     @InternalOnly
     public ConsumptionExistenceResponse doesStorageConsumptionCollectionExist(@AccountId String accountId,
             @NotNull @ValidCrn(resource = {CrnResourceDescriptor.ENVIRONMENT, CrnResourceDescriptor.DATALAKE}) String monitoredResourceCrn,
-            @NotEmpty String storageLocation, @ValidCrn(resource = { CrnResourceDescriptor.USER, CrnResourceDescriptor.MACHINE_USER })
-            @InitiatorUserCrn @NotEmpty String initiatorUserCrn) {
+            @NotEmpty String storageLocation,
+            @ValidCrn(resource = { CrnResourceDescriptor.USER, CrnResourceDescriptor.MACHINE_USER }) @InitiatorUserCrn @NotEmpty String initiatorUserCrn) {
+        return getConsumptionExistenceResponse(monitoredResourceCrn, storageLocation);
+    }
+
+    @Override
+    @InternalOnly
+    public void scheduleConsumptionCollection(@AccountId String accountId,
+            @Valid @NotNull StorageConsumptionRequest request,
+            @ValidCrn(resource = {CrnResourceDescriptor.USER, CrnResourceDescriptor.MACHINE_USER}) @NotEmpty String initiatorUserCrn) {
+        LOGGER.info("Registering consumption collection for resource with CRN [{}] and location [{}]",
+                request.getMonitoredResourceCrn(), request.getStorageLocation());
+        scheduleConsumption(request, request.getConsumptionType());
```

Comment:
if getConsumptionType == null should we set to storage as default ?

Reply:
Yes we should.Corrected